### PR TITLE
2922811 - Deprecate Commerce's BundleFieldDefinition, BundlePluginInterface

### PR DESCRIPTION
**Pull request by bradjones1**
https://www.drupal.org/project/issues/commerce/2922811

### CODE REVIEW

File: src/BundleFieldDefinition.php [hunk 1]
```diff
@@ -2,11 +2,15 @@
 
 namespace Drupal\commerce;
 
+@trigger_error('The '. __NAMESPACE__ . '\BundleFieldDefinition is deprecated. Instead, use \Drupal\entity\BundleFieldDefinition', E_USER_DEPRECATED);
```
Comment (bojanz) [inline src/BundleFieldDefinition.php]: ----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 5 | ERROR | [x] Concat operator must be surrounded by a single space
   |       |     (Squiz.Strings.ConcatenationSpacing.PaddingFound)

Comment (bradjones1) [reply src/BundleFieldDefinition.php]: Fixed. Also figured we should do the same with `BundlePluginInterface`.

**Comment by bradjones1** [comment]
Rebased.

**Comment by bojanz** [comment]
Unfortunately, this will immediately crash all of contrib that uses bundle plugins (recurring, license, gateways with custom payment method types), because ConfigurableFieldManager gets an unexpected class (\Drupal\commerce\BundleFieldDefinition and \Drupal\entity\BundleFieldDefinition are not interchangeable). And if we keep the old typehint in ConfigurableFieldManager, then there's no way to move off the deprecated classes.

So it looks like deprecating BundlePluginInterface is actually the only part we can do.

EDIT: Nevermind, looks like I'm wrong, the Commerce BundleFieldDefinition extends the Entity one. The crash I observed was without this PR, trying to change the commerce_recurring usage before Commerce is fixed.

**Comment by bojanz** [comment]
Committed in 654bf3db4c1a879193ab4437e0865d2b69c74ccf.

**Pull request closed without merging**